### DISCUSSION TITLE
Hight CPU usage on handle count executed from powershell

### DISCUSSION THREAD

**Discussion by sturla78** [rabbitmq/rabbitmq-server#8541] [Other]
### Describe the bug

Hello,
i have similar issue that i have posted some time ago but now regards the new logic present in rabbit 3.11.17 (erlang 25.3.2.1) on windows when is performed a check on the handle count, the command executed is the following:

c:/windows/system32/windowspowershell/v1.0/powershell.exe -NoLogo -NonInteractive -NoProfile -InputFormat Text -OutputFormat Text -Command "Get-Process -Id 3516 | Select-Object -ExpandProperty HandleCount"

This command is performed closely and on some machine can occupy a handful of seconds and the cpu is very high during this moments. This happen with windows 2019 server and powershell version  5.1.17763.3770 but not happen on windows 2022  and powershell version  5.1.20348.643 (with this s.o. the powershell execution is normal). 
I have also attached an image with the cpu behavior on both system.

Thanks in advance for the reply.

![handle count](https://github.com/rabbitmq/rabbitmq-server/assets/5145973/3a945c5f-f256-453f-8324-863d0cfa8798)

### Reproduction steps

1. Install rabbitmq 3.11.17 with erlang 25.3.2.1 on a windows 2019 machine
2. Open a process explorer tool and verify the cpu occupation during the execution of the command that count the handle

### Expected behavior

There is a solution in order to mitigate this problem with powershell ?

**Comment by michaelklishin** [comment]
If you know of a less resource-intensive alternative, you are welcome to share it. I do not see anything particularly demanding or unusual in this shell out to PowerShell in non-interactive mode.

We obtain information about a single process and not all processes, to then filter them down to just one. I do not have any hypothesis as to what may be so CPU extensive. Perhaps your machine has really limited CPU resources, such as a single core.

**Comment by lukebakken** [comment]
> i have similar issue that i have posted some time ago

What issue and where did you post it?

----

In previous RabbitMQ versions, you would get a warning log message that `handle.exe` is missing. Now, we use PowerShell as a backup.

If you'd like to fix this issue, please ensure Sysinternal's `handle.exe` tool is installed and in the PATH on all of your RabbitMQ servers - https://learn.microsoft.com/en-us/sysinternals/downloads/handle

**Reply by sturla78** [reply]
> > i have similar issue that i have posted some time ago
> 
> What issue and where did you post it?
> 
> In previous RabbitMQ versions, you would get a warning log message that `handle.exe` is missing. Now, we use PowerShell as a backup.
> 
> If you'd like to fix this issue, please ensure Sysinternal's `handle.exe` tool is installed and in the PATH on all of your RabbitMQ servers - https://learn.microsoft.com/en-us/sysinternals/downloads/handle

I was referring to free disk space that has been resolved with rabbitmq 3.10.X, changing the method from powershell with erlang executable.
https://github.com/rabbitmq/rabbitmq-server/issues/4140#issuecomment-1174831067

Sorry, a last question: in this case there is no the possibility to disable this behaviour ? The only way is to configure handle.exe ? 

Thanks again

**Reply by lukebakken** [reply]
> Sorry, a last question: in this case there is no the possibility to disable this behaviour ? The only way is to configure handle.exe ?

Yes, installing `handle.exe` is your only option.

**Comment by lukebakken** [comment]
@sturla78 please watch this issue as well - https://github.com/rabbitmq/rabbitmq-server/issues/8700

**Reply by sturla78** [reply]
Thanks @lukebakken, 
i'll be monitoring the discussion.

**Reply by lukebakken** [reply]
Let us know if installing `handle.exe` does _not_ fix the issue for you.

**Reply by sturla78** [reply]
I can confirm that installing handle this _resolve the issue_, seems that the priority, with the presence of both tools, is for handle and in this case powershell is not used.

**Comment by neoneo40** [comment]
Sorry.
Originally, I was using RabbitMQ in Windows, so I kept using it, but the CPU usage rate soared so I had no choice but to change it.
I changed message broker server form Rabbit MQ to Redis.
I used to install handle.exe but not working for me. Same CPU high usage
CPU usage: 90% -> 40%

**Reply by lukebakken** [reply]
> I used to install handle.exe but not working for me. Same CPU high usage

Just to clarify, you installed `handle.exe`, but the CPU usage stayed at 90%?

After installing `handle.exe`, did you restart RabbitMQ?
